readiness-tool/popup.js

```diff
@@ -0,0 +1,218 @@
+};
+/**
+ * Returns all the 3rd party applications found on the website
+ * @param {String} html - String containing all HTML on the page
+ * @return {Object} 
+ */
+function findDetectedApps(html) {
+  fetch('apps.json').then(function (response) {
+    if (response.status !== 200) {
+      console.log('Looks like there was a problem. Status Code: ' + response.status);
+      return;
+    }
+    // Examine the text in the response  
+    response.json().then(function (data) {
+      listAllApps = data.apps;
+      detectedApps = filterApps(html);
+      showSupportedAppsInView(detectedApps);
+      return detectedApps;
+    });
+  }).catch(function (err) {
+    console.log('Fetch Error :-S', err);
+  });
+}
+/**
+ * Add supported and unsupported applications to the view
+ * @param {Objected} detectedApps - All 3rd Party Applications found on page
+ */
+function showSupportedAppsInView(detectedApps) {
+  supportedAds.innerHTML = supportedAnalytics.innerHTML = notSupportedAds.innerHTML = notSupportedAnalytics.innerHTML = blankMessage;
+  supportedAds.appendChild(makeList(detectedApps.supported.ads, false));
+  supportedAnalytics.appendChild(makeList(detectedApps.supported.analytics, false));
+  notSupportedAds.appendChild(makeList(detectedApps.notSupported.ads, true));
+  notSupportedAnalytics.appendChild(makeList(detectedApps.notSupported.analytics, true));
+}
+/**
+ * Splits all detected apps into 'supported' and 'not supported'
+ * @param {String} htmlString - String containing all HTML on the page
+ * @return {Object} 
+ */
+function filterApps(htmlString) {
+  const foundThis = {
+    'supported': {
+      'ads': [],
+      'analytics': []
+    },
+    'notSupported': {
+      'ads': [], 
+      'analytics': []
+    }
+  };
+  // for all the app objects in the apps.JSON file
+  Object.keys(listAllApps).forEach(function (key) {
+    let val = listAllApps[key];
+    // If object has a 'script' key
+    if (val.script != null) {
+      Object.keys(val.script).forEach(function (x) {
+        addToDict(val.script[x].split('\\;')[0], htmlString, foundThis, key, val.cats);
+      });
+    }
+  });
+  return foundThis;
+}
+/**
+ * Pushes keys to the supported or not supported list of the object 'found this'
+ * @param {String} tempScript - String representation of regular expression
+ * @param {!String} htmlString - String containing all HTML on the page
+ * @param {Object} foundThis - Object separating the 3P services by support
+ * @param {String} key - name of third party service
+ * @param {String} category - the category that the key belongs to
+ */
+function addToDict(tempScript, htmlString, foundThis, key, category) {
+  const regX = new RegExp(tempScript);
+  if (regX.test(htmlString)) {
+    if (isKeyUnique(foundThis, key)) {
+      switch (true) {
+      case (isSupported(key) == true) && category == 10:
+        foundThis.supported.analytics.push(key);
+        break;
+      case (isSupported(key) == true) && category == 36:
```

Comment:
Leave out `== true` or `== false` since it's superfluous.

```js
// Don't do this.
if (someBoolean == true) {
}

// Do this instead.
if (someBoolean) {
}
```